### PR TITLE
nixVersions.unstable: 2.17 -> 2.18.1

### PR DESCRIPTION
**Pull request by colemickens**
## Description of changes

Backstory:

* Nixpkgs reverted `nixUnstable` from `2.18` to `2.17.x.`: https://github.com/NixOS/nixpkgs/pull/258432
* due to this issue: https://github.com/NixOS/nix/issues/9052
* which was fixed in: https://github.com/NixOS/nix/pull/9081
* backported in: https://github.com/NixOS/nix/pull/9086
* and is available in `2.18.1`

and we already bumped `nix_2_18` to `2.18.1`.

It seems like (maybe) it's the responsibility of the @NixOS/nix-team to bump stable. I'm not sure under what conditions that's done, but now that nix#9052 is fixed, it seems we should bump unstable again.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fricklerhandwerk** [comment]
IMO Nixpkgs maintainers should choose the Nix version they want to rely on.

**Comment by colemickens** [comment]
@fricklerhandwerk Can you clarify what you mean by "Nixpkgs maintainers"? Maybe you mean the maintainers listed in `pkgs/tools/package-management/nix/common.nix`? If so, I'm happy to request reviews from them.

Or maybe you mean the @NixOS/nix-team ? I tried asking about policy, etc in `#dev:matrix.org` and didn't get much of a firm answer other than semver-minor rev bumps for stable should be handled by @NixOS/nix-team.

**Review by Artturin — APPROVED** [review]
The reason why was fixed so 👍

**Comment by fricklerhandwerk** [comment]
> Maybe you mean the maintainers listed in `pkgs/tools/package-management/nix/common.nix`?

At the very least. For the 23.05 release it felt like @infinisil and the Nixpkgs Architecture Team as well as @RaitoBezarius and the @NixOS/release-engineers (or @NixOS/nixos-release-managers? Why are there two teams?) were primarily involved in that decision.

Nix maintainers should be consulted on a pull basis only when needed.

**Comment by grahamc** [comment]
> IMO Nixpkgs maintainers should choose the Nix version they want to rely on.

Looks like @colemickens is one of the many Nixpkgs maintainers, and this is a bump for the unstable Nix. What's the scoop here? Why wouldn't we put 2.18.1 in nixpkgs? On the contrary it seems quite stable from my perspective, and I think it would be ideal to make the stable Nix 2.18.x before branchoff.

**Comment by RaitoBezarius** [comment]
I think the only concern is for nix stable.

**Comment by grahamc** [comment]
Sounds good. Since it got a +1 from a package maintainer, let's merge.

**Pull request merged**
